extensions/cloudlink.js

```diff
@@ -1,1838 +1,2406 @@
-            blockAllThreads: "true",
+            blockType: Scratch.BlockType.REPORTER,
+            hideFromPalette: clVars.hideCLDeprecatedBlocks,
             text: "Fetch data from URL [url]",
             arguments: {
               url: {
-                type: "string",
+                type: Scratch.ArgumentType.STRING,
                 defaultValue: "https://extensions.turbowarp.org/hello.txt",
               },
             },
           },
+
           {
             opcode: "requestURL",
-            blockType: "reporter",
-            blockAllThreads: "true",
+            blockType: Scratch.BlockType.REPORTER,
+            hideFromPalette: clVars.hideCLDeprecatedBlocks,
             text: "Send request with method [method] for URL [url] with data [data] and headers [headers]",
             arguments: {
               method: {
-                type: "string",
+                type: Scratch.ArgumentType.STRING,
                 defaultValue: "GET",
               },
               url: {
-                type: "string",
+                type: Scratch.ArgumentType.STRING,
                 defaultValue: "https://extensions.turbowarp.org/hello.txt",
               },
               data: {
-                type: "string",
+                type: Scratch.ArgumentType.STRING,
                 defaultValue: "{}",
               },
               headers: {
-                type: "string",
+                type: Scratch.ArgumentType.STRING,
                 defaultValue: "{}",
               },
             },
           },
-          {
-            opcode: "makeJSON",
-            blockType: "reporter",
-            text: "Convert [toBeJSONified] to JSON",
-            arguments: {
-              toBeJSONified: {
-                type: "string",
-                defaultValue: '{"test": true}',
-              },
-            },
-          },
+
+          "---",
+
           {
             opcode: "onConnect",
-            blockType: "hat",
+            blockType: Scratch.BlockType.EVENT,
             text: "When connected",
-            blockAllThreads: "true",
+            isEdgeActivated: false, // Gets called by runtime.startHats
+            shouldRestartExistingThreads: true,
           },
+
           {
             opcode: "onClose",
-            blockType: "hat",
+            blockType: Scratch.BlockType.EVENT,
             text: "When disconnected",
-            blockAllThreads: "true",
+            isEdgeActivated: false, // Gets called by runtime.startHats
+            shouldRestartExistingThreads: true,
           },
+
+          "---",
+
           {
             opcode: "onListener",
-            blockType: "hat",
-            text: "When I receive new packet with listener [ID]",
-            blockAllThreads: "true",
+            blockType: Scratch.BlockType.HAT,
+            text: "When I receive new message with listener [ID]",
+            isEdgeActivated: true, // Set to false when runtime.startHats is fixed
```

Comment:
what issue?

Reply:
I kept receiving errors relating to a runtime/VM issue. Something similar to a .value error, as well as a "not a function" error outside the scope of my extension code.

Reply:
That error would imply you passed in an object like `{something: undefined}` as the second argument, which isn't valid, the values should just be strings. If you don't want to filter by a field then just don't list it there.